include/aspect/postprocess/particle_distribution_statistics.h

```diff
@@ -0,0 +1,144 @@
+#include <deal.II/base/patterns.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that generates a point-density function describing
+     * the distribution of particles within cells. The point-density function
+     * is generated using a Kernel Density Estimator. The postprocessor calculates
+     * standard deviation and max/min of the point-density functions per cell.
+     *
+     * @ingroup Postprocessing
+     */
+    template <int dim>
+    class ParticleDistributionStatistics : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Evaluate the solution for some particle statistics.
+         */
+        std::pair<std::string,std::string>
+        execute (TableHandler &statistics) override;
+
+        /**
+         * Let the postprocessor manager know about the other postprocessors
+         * this one depends on. Specifically, the particles postprocessor.
+         */
+        std::list<std::string>
+        required_other_postprocessors() const override;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        void
+        parse_parameters (ParameterHandler &prm) override;
+      private:
+        /**
+         * If `KDE_per_particle` is true, the point-density function is defined
+         * at the position of every particle in the cell. If it is false, the
+         * point-density-function is defined on a regular grid throughout the cell.
+         */
+        bool KDE_per_particle;
+
+        /**
+         * The `granularity` variable determines the number of points at which
+         * the point-density function is defined within each cell. For example,
+         * a value of 2 means that the point-density function is defined at
+         * $2\times 2=4$ points in 2D. This variable only applies if
+         * `KDE_per_particle` is false.
+         */
+        unsigned int granularity;
+
+        /**
+         * The `bandwidth` variable scales the point-density function.
+         * Choosing an appropriate bandwidth is important because a
+         * bandwidth value which is too low or too high can result
+         * in oversmoothing or undersmoothing of the point-density function.
+         * Oversmoothing or undersmoothing results in a function which
+         *  represents the underlying data less accurately.
+         */
+        double bandwidth;
+
+        /**
+         * The `KernelFunctions` enum class is a data structure which
+         * contains the kernel functions available for use in the Kernel
+         * Density Estimator.
+         */
+        enum class KernelFunctions
+        {
+          GAUSSIAN,
+          TRIANGULAR,
+          UNIFORM,
+          CUTOFF_FUNCTION_W1_DEALII
```

Comment:
Let's spell these in lower-case. They're members of an enum class, so they have a `KernelFunctions::` prefixed and are therefore just regular variables/values.